src/hotspot/share/gc/shared/space.inline.hpp

```diff
@@ -183,7 +186,8 @@ inline void CompactibleSpace::scan_and_forward(SpaceType* space, CompactPoint* c
       // we don't have to compact quite as often.
       if (cur_obj == compact_top && dead_spacer.insert_deadspace(cur_obj, end)) {
         oop obj = oop(cur_obj);
-        compact_top = cp->space->forward(obj, obj->size(), cp, compact_top);
+        size_t obj_size = obj->size();
+        compact_top = cp->space->forward(obj, obj_size, cp, compact_top);
```

Comment:
This change seems superfluous now. Inline `obj_size` back?

Reply:
Done